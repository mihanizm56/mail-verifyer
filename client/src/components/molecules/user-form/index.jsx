import React, { Component } from "react";
import { Formik, Field, Form, ErrorMessage, FieldArray, withFormik } from "formik";
import { withTranslation } from "react-i18next";

export class WrappedForm extends Component {
	componentDidUpdate(prevProps) {
		const { error, setErrors } = this.props;

		if (error && prevProps.error !== error) {
			setErrors({ name: error, email: error });
		}
	}

	render = () => {
		const { values, t: translate } = this.props;

		return (
			<div>
				<Form>
					<FieldArray
						name="email-sender"
						render={() => (
							<div>
								<div className="col">
									<label htmlFor="name">{translate("mail.send-form.name")}</label>
									<Field name="name" value={values.name} type="text" />
									<ErrorMessage name="name" component="div" className="field-error" />
								</div>
								<div className="col">
									<label htmlFor="email">{translate("mail.send-form.email")}</label>
									<Field name="email" value={values.email} type="email" />
									<ErrorMessage name="email" component="div" className="field-error" />
								</div>
							</div>
						)}
					/>
					<button type="submit">{translate("mail.send-form.validate")}</button>
				</Form>
			</div>
		);
	};
}

// TODO write a regular composition
export const FormikUserForm = withFormik({
	handleSubmit: (values, { props }) => {
		props.submitFunc(values);
	},
	mapPropsToValues: props => {
		return {
			name: "test",
			email: "test@mail.ru",
		};
	},
})(WrappedForm);

export const UserForm = withTranslation()(FormikUserForm);
