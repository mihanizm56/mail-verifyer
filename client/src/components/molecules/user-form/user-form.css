.user-form {
    border: 1px solid green;
    border-radius: 20px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 800px;
    min-height: 650px
}

.user-form__row {
    width: 80%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 40px 40px;
}

.user-form__row--button {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.user-form__field-block {
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.user-form__text-field {
    font-size: var(--font-size-error);
    color: var(--black);
    padding: 0 10px;
    width: 250px;
    height: 50px;
    border-radius: 10px;
    border: 1px solid grey;
    outline: none;
    transition: .2s;
}

.user-form__text-field:focus {
    box-shadow: 0 0 0 2pt grey;
}

.user-form__error {
    height: 30px;
}

.user-form__error-text {
    text-align: center;
    font-size: var(--font-size-error);
    color: var(--color-error-text)
}

@media screen and (max-width: 1199px) and (min-width: 768px) {
    .user-form {
        width: 90%;
    }
}

@media screen and (max-width: 767px) {
    .user-form {
        width: 95%;
    }
    .user-form__row:first-child {
        padding: 40px 40px;
    }
    .user-form__row {
        padding: 20px 20px;
    }
    .user-form__field-block {
        padding-top: 10px;
        flex-direction: column;
    }
    .user-form__error {
        height: 30px;
    }
    .user-form__text-field {
        order: 1;
    }
    .user-form__label {
        order: 2;
        display: flex;
        justify-content: center;
    }
}