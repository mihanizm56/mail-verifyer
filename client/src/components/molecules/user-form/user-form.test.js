import React from "react";
import { mount } from "enzyme";
import { UserForm, WrappedForm } from "../user-form";

describe("test", () => {
	it("test snap", () => {
		const props = {
			values: {
				username: "",
				email: "",
			},
			isLoading: false,
			submissionError: null,
			errors: null,
			t: () => "translated",
		};
		const component = mount(<UserForm {...props} />).find("user-form");

		expect(component).toMatchSnapshot();
	});
});
