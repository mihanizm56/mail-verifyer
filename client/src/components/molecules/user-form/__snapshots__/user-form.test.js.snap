// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test snap 1`] = `null`;
